src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -160,8 +219,19 @@ public override void Read(Span<byte> data)
         /// <inheritdoc />
         public override void Write(ReadOnlySpan<byte> data)
         {
-            Span<byte> dataToRead = stackalloc byte[data.Length];
-            TransferFullDuplex(data, dataToRead);
+            if (_miso == -1)
```

Comment:
Since you already have TransferWriteOnly I'd just always use it

Reply:
I didn't want to change too much, but I guess the devices won't know if the data is read or not. Let me change it.